lib/rqrcode/qrcode/qr_code.rb

```diff
@@ -117,7 +129,8 @@ def initialize( string, *args )
 
       @data                 = string
 
-      mode                  = QRAlphanumeric.valid_data?( @data ) ? :mode_alpha_numk : :mode_8bit_byte
+      mode                  = QRMODE_NAME[(options[:mode] || '').to_sym]
+      mode                  ||= QRAlphanumeric.valid_data?( @data ) ? :mode_alpha_numk : :mode_8bit_byte  # deprecate?
```

Comment:
I think automatic is great. But I really like that you added the possibility to set mode explicit.